File: app/src/main/java/dev/ragnarok/fenrir/settings/AppPrefs.java
package dev.ragnarok.fenrir.settings;

import android.content.Context;
import android.content.pm.PackageManager;

public class AppPrefs {

    public static boolean isCoubInstalled(Context context) {
        return isPackageIntalled(context, "com.coub.android");
    }

    public static boolean isNewPipeInstalled(Context context) {
        return isPackageIntalled(context, "org.schabi.newpipe");
    }

    public static boolean isYoutubeInstalled(Context context) {
        return isPackageIntalled(context, "com.google.android.youtube");
    }

    public static boolean isVancedYoutubeInstalled(Context context) {
        return isPackageIntalled(context, "com.vanced.android.youtube");
    }

    private static boolean isPackageIntalled(Context context, String name) {
        PackageManager pm = context.getPackageManager();
        try {
            pm.getPackageInfo(name, PackageManager.GET_ACTIVITIES);
            return true;
        } catch (PackageManager.NameNotFoundException ignored) {
            return false;
        }
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/settings/AppPrefs.kt
package dev.ragnarok.fenrir.settings

import android.content.Context
import android.content.pm.PackageManager

object AppPrefs {
    fun isCoubInstalled(context: Context): Boolean {
        return isPackageIntalled(context, "com.coub.android")
    }

    fun isNewPipeInstalled(context: Context): Boolean {
        return isPackageIntalled(context, "org.schabi.newpipe")
    }

    fun isYoutubeInstalled(context: Context): Boolean {
        return isPackageIntalled(context, "com.google.android.youtube")
    }

    fun isVancedYoutubeInstalled(context: Context): Boolean {
        return isPackageIntalled(context, "com.vanced.android.youtube")
    }

    private fun isPackageIntalled(context: Context, name: String): Boolean {
        val pm = context.packageManager
        return try {
            pm.getPackageInfo(name, PackageManager.GET_ACTIVITIES)
            pm.getApplicationInfo(name, 0).enabled
        } catch (ignored: PackageManager.NameNotFoundException) {
            false
        }
    }
}